backend/routes/seatbid.js

```diff
@@ -121,8 +135,25 @@ router.post('/accept', authenticateUser, async (req, res) => {
             }),
         ]);
 
-        activeBiddings.delete(biddingId);
-        bids.delete(biddingId);
+        seatsToRemove.forEach(seatNumber => {
+            const biddingIdToRemove = `${bidding.flightId}-${seatNumber}`;
+            activeBiddings.delete(biddingIdToRemove);
+            bids.delete(biddingIdToRemove);
+        });
+
+        activeBiddings.forEach((activeBidding, activeBiddingId) => {
+            if (!seatsToRemove.includes(activeBidding.seatNumber)) {
```

Comment:
@nowshin-108 Could you explain this logic? If I understand correctly, we are removing bids associated to the seat that was just accepted, right?

Quick question, do we allow users creating a bidding to also bid in other biddings?

- If not, can we do that?
- If yes, shall we invalidate their biddings if their seat was accepted in a separate bidding?

Feel free to implement this in separate PR.

Reply:
Yeap, users do can create a bidding for their seat and also bid on other seats simultaneously. If a user's bid is accepted in a separate bidding, their original bidding and all their bids on other seats are automatically invalidated. 

`        seatsToRemove.forEach(seatNumber => {
            const biddingIdToRemove = `${bidding.flightId}-${seatNumber}`;
            activeBiddings.delete(biddingIdToRemove);
            bids.delete(biddingIdToRemove);
        });

        activeBiddings.forEach((activeBidding, activeBiddingId) => {
            if (!seatsToRemove.includes(activeBidding.seatNumber)) {
                const updatedBids = bids.get(activeBiddingId).filter(bid => 
                    !seatsToRemove.includes(bid.bidderSeatNumber)
                );
                bids.set(activeBiddingId, updatedBids);

                if (updatedBids.length === 0) {
                    activeBiddings.delete(activeBiddingId);
                    bids.delete(activeBiddingId);
                }
            }
        });`                                                                                                                       

This code basically removes the biddings and bids for the seats directly involved in the swap. Then, it iterates through all other active biddings, removing any bids made from the swapped seats and deleting biddings that end up with no bids. This is to ensure system-wide consistency after a seat swap.